src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -139,11 +139,18 @@ public void run() {
             throw new RuntimeException("READER ERROR");
         } finally {
             try {
-                conn.close();
+                shutdown();
+                LOG.error("Connection to the database was closed.");
             } catch (Exception e) {
-                LOG.error("Unable to close database connection.");
+                LOG.error("Unable to close all database connections and shutdown cleanly.");
             }
-            LOG.error("Connection to the database was closed.");
+        }
+    }
+
+    public void shutdown() throws Exception {
+        conn.close();
+        for (MetricsDB db: metricsDBMap.values()) {
+            db.close();
```

Comment:
If exception on any db.close() or conn.close(), all the remaining connections will not be closed. Is it better to absorb the close() exception here and continue for remaining connections? Or, any other thoughts for doing this way

Reply:
Good point. Will change it.